packages/ssx-server/src/middlewares/express/middleware.ts

```diff
@@ -88,16 +88,14 @@ export const ssxMiddleware = (ssx: SSXServer) => {
 
     if (req.session?.siwe) {
       const { signature, siwe, daoLogin, nonce } = req.session;
-      let siweMessageVerify;
-      try {
-        siweMessageVerify = await new SiweMessage(siwe).verify(
-          { signature, nonce },
-          {
-            verificationFallback: daoLogin ? SiweGnosisVerify : null,
-            provider: ssx.provider,
-          },
-        );
-      } catch (error) {}
+      const siweMessageVerify = await new SiweMessage(siwe).verify(
+        { signature, nonce },
+        {
+          verificationFallback: daoLogin ? SiweGnosisVerify : null,
+          suppressExceptions: true,
```

Comment:
I think we should avoid using exceptions and try/catch blocks in our codebase. In building our Typescript SDK we should be able to provide the reasons for failures and "exceptions" to expected behaviors and failure modes should probably be exceptional. There are also performance and type-safety losses as well

Reply:
I think this is not something we want to do https://github.com/spruceid/siwe/blob/053f16bf7dce3ff587e00bc4d3754a29ee9c26e6/packages/siwe/lib/client.ts#L195-L201. We shouldn't swallow errors in the first place - a failed promise without an error handler is def an exception and should throw. If the goal here is to avoid try/catch and `SiweMessage#verify` returns a `Promise`, then just use `then(onFulfilled, onReject)`.

Reply:
Addressed